apps/theming/lib/Themes/CommonThemeTrait.php

```diff
@@ -0,0 +1,65 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2022 Joas Schilling <coding@schilljs.com>
+ *
+ * @author Joas Schilling <coding@schilljs.com>
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\Theming\Themes;
+
+use OCA\Theming\Util;
+
+trait CommonThemeTrait {
+	public Util $util;
+
+	/**
+	 * Generate primary-related variables
+	 * This is shared between multiple themes because colorMainBackground and colorMainText
+	 * will change in between.
+	 */
+	protected function genPrimaryVariables(string $colorMainBackground, string $colorMainText): array {
```

Comment:
```suggestion
	protected function generatePrimaryVariables(string $colorMainBackground, string $colorMainText): array {
```

we wouldn't need to save characters like this

Reply:
Done